Admin UI and minor fixes

marsian83 opened the pull request:
> Marked as draft for now

marsian83 commented:
@frouioui  please review, I have added client side functionality for request and delete from queue

frouioui commented:
Hey @marsian83 thank you for this Pull Request.

frouioui commented on website/src/pages/DailyPage/DailyPage.tsx:
When reaching `http://localhost/daily` the page loads for a little while and then a black screen is rendered. I can see in the console the following error message:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'map')
at DailySummary (DailySummary.tsx:35:23)
```

frouioui commented on website/src/common/Macrobench.tsx:
When reaching `http://localhost/macro`, same as the Daliy page, the page loads and then a black screen appears with some errors in the console:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'CommitHash')
```

frouioui commented on website/src/common/AdminSideNav.tsx:
```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useRef, useState } from "react";
+import { NavLink } from "react-router-dom";
```
Header is missing

frouioui commented on website/src/common/AdminSideNav.tsx:
```diff
@@ -0,0 +1,91 @@
+  { title: "Dashboard", to: "/", icon: "person" },
+  { title: "Benchmarks", to: "/benchmarks", icon: "ssidChart" },
+  { title: "Logs", to: "/logs", icon: "description" },
+  // { title: "Dummy", to: "/dummy", icon: "key" },
```
let's remove this line if unused

frouioui commented on website/src/common/ThemeToggleButton.tsx:
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { twMerge } from "tailwind-merge";
```
Header is missing here too

frouioui commented on website/src/hooks/useApiCall.ts:
```diff
@@ -0,0 +1,42 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
```
Here as well

frouioui commented on website/src/hooks/useFetch.ts:
```diff
@@ -0,0 +1,36 @@
+import axios, { AxiosRequestConfig } from "axios";
+import { useEffect, useState } from "react";
```
Here too

frouioui commented on website/src/hooks/usePromisedState.ts:
```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from "react";
+
```
Header

frouioui commented on website/src/pages/AdminBenchmarksPage/AdminBenchmarksPage.tsx:
```diff
@@ -0,0 +1,85 @@
+import admin from "../../utils/admin";
+import { twMerge } from "tailwind-merge";
+
+const benchmarkWorkloads = [
+  "OLTP-SET",
+  "OLTP-READONLY",
+  "OLTP-READONLY-OLAP",
+  "TPCC",
+  "OLTP",
+];
```
Can we make this list dynamic based on the list of benchmarks returned by the API?

harry-sandhu replied:
> Can we make this list dynamic based on the list of benchmarks returned by the API?

I thnk I know how to do this
```
func (s *Server) getBenchmarkNames(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"benchmarks": s.benchmarkTypes,
	})
}
```

This code snippet if added in `go/server` and adding a respective route like `/api/get-benchmark-types`
Will give list which we can use in frontend.
@frouioui

frouioui commented on website/src/pages/AdminBenchmarksPage/components/ExecutionQueue.tsx:
```diff
@@ -0,0 +1,96 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import React, { useEffect, useState } from "react";
```
```suggestion
*/

import React, { useEffect, useState } from "react";
```

frouioui commented on website/src/pages/AdminPanelPage/AdminPanelPage.tsx:
```diff
@@ -0,0 +1,8 @@
+
+export default function AdminPanelPage() {
```
Header is missing here too

frouioui commented on website/src/pages/ErrorPage/ErrorPage.tsx:
```diff
@@ -0,0 +1,19 @@
+import { Link } from "react-router-dom";
+
```
Header  here too :)

frouioui commented on website/src/pages/ErrorPage/ErrorPage.tsx:
```diff
@@ -0,0 +1,19 @@
+      <div className="w-[20vw] aspect-square translate-x-[11%]">
+        <img src="/404.png" alt="error" />
+      </div>
+      <h2>We couldn't find he page you were looking for</h2>
```
```suggestion
      <h2>We could not find the page you are looking for</h2>
```

frouioui commented on website/src/pages/TestingPage/TestingPage.tsx:
```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from "react";
```
header missing

frouioui commented on website/src/pages/TestingPage/TestingPage.tsx:
```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from "react";
+// import api from "../../services/api";
+
```
let's remove this

frouioui commented on website/src/types/Data.ts:
```diff
@@ -0,0 +1,194 @@
+import Macrobench from "../common/Macrobench";
+import { BenchmarkStatus } from "./enums";
```
header

frouioui commented on website/src/types/Data.ts:
```diff
@@ -0,0 +1,194 @@
+  Result: MicroResult;
+}
+
+export interface Macros {
+  OLTP: MacroBenchmark[] | null;
```
same thing here as mentioned above, would it be possible to dynamically load the name of the macro benchmarks? the list of macro benchmark evolve over time

codingmickey commented:
Hey @frouioui @marsian83 does this need any help, I'd be happy to take this up and bring it to at least a completion!

frouioui commented:
@codingmickey Hello! I don't think @marsian83 has planned on continuing this given there was no response for months. You're welcome to take this on!

marsian83 commented:
> Hey @frouioui @marsian83 does this need any help, I'd be happy to take this up and bring it to at least a completion!

Hi, it'd be great if you could honestly.
The majority of the issues currently are due to mismatch of the type declarations we have on the frontend side vs the responses we recieve from the server. 
I'd recommend a straigh forward approach of checking the response types for the api calls in the errored pages and changing around the types accordingly.
If you need any help understanding anything do let me know.

marsian83 commented:
Hi @codingmickey 
I see there are a lot of conflicts with the current state of the HEAD and the main
let me resolve these conflicts first before you could look into this
Mind waiting a day or two?

codingmickey commented:
Sure anyways I have viva exam today and on Monday 😅

On Fri, 10 May, 2024, 10:46 pm Spandan Barve, ***@***.***>
wrote:

> Hi @codingmickey <https://github.com/codingmickey>
> I see there are a lot of conflicts with the current state of the HEAD and
> the main
> let me resolve these conflicts first before you could look into this
> Mind waiting a day or two?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vitessio/arewefastyet/pull/469#issuecomment-2104983221>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKEMS6Y3PN2PTS3W5XEA4R3ZBT6H3AVCNFSM6AAAAAA6MSXZQWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMBUHE4DGMRSGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

harry-sandhu commented:
Hi, I made progress for this
I fix some pages, where api had changed so I changed the api.ts file
and fixed the rendering methods where the api response was being used
But I did that on @marsian83, your fork's head
but I can not push

@frouioui @marsian83 Can you please guide me aroudn this github issue?

marsian83 commented:
> Hi, I made progress for this I fix some pages, where api had changed so I changed the api.ts file and fixed the rendering methods where the api response was being used But I did that on @marsian83, your fork's head but I can not push
> 
> @frouioui @marsian83 Can you please guide me aroudn this github issue?

You can fork my branch and then create PR to it, when I merge those, it will be added here.
have you already made commits?

harry-sandhu commented:
I have fixed all pages
Daily page is not working, the endpoint returns `null`
the endpoint `/api/daily` even the production version `https://benchmark.vitess.io/api/daily` returns null
Please help me with this @frouioui  @Camillemtd @marsian83  
Also, I request you to test and review 
.Thanks

marsian83 commented on website/src/common/Macrobench.tsx:
```diff
@@ -138,73 +138,73 @@ export default function Macrobench(props: MacrobenchProps) {
             right={fixed(data.diff.Right.Result.threads, 2)}
             diffMetric={fixed(data.diff.Diff.threads, 2)}
-          />
+          /> */}
```
Why

harry-sandhu replied:
the response returned by api  does not have specific values that correspond to these .This is the reason they are commented out.

marsian83 commented on website/src/hooks/useApiCall.ts:
```diff
@@ -23,6 +23,7 @@ export default function useApiCall<T extends ApiEndpoint>(
         },
         ...config,
       });
+      console.log(`Response from ${uri}:`, response.data);
```
Please remove any unused / unnecessary console logs

marsian83 commented on website/src/pages/SearchPage/components/SearchMacro.tsx:
```diff
@@ -20,11 +20,11 @@ import PropTypes from "prop-types";
 import { Link } from "react-router-dom";
 
-export default function SearchMacro({ data, gitRef }: any) {
+export default function SearchMacro({ data, macroName, gitRef }: any) {
```
Can you figure out how to do this without explicitly using `any`

marsian83 commented:
I have reviewed your code @harry-sandhu 
Testing seems to indicate all pages work fine, except of course the daily page. 
I will see what can be wrong with the `/api/daily` endpoint

frouioui commented:
I will close this as @Jad31 is going to do the admin UI again.
